### DISCUSSION TITLE
Document capability support in a scalable way

### DISCUSSION THREAD

**Discussion by beriberikix** [zephyrproject-rtos/zephyr#35981] [Ideas]
It is somewhat hard to identify what capabilities a SOC/board supports in a consistent and centralized way. By consistent I mean different boards document supported features at different levels of accuracy and some don't document them at all. For example, see the [nRF52840DK](https://docs.zephyrproject.org/latest/boards/arm/nrf52840dk_nrf52840/doc/index.html#supported-features) vs the [ESP32](https://docs.zephyrproject.org/latest/boards/xtensa/esp32/doc/index.html). Relatedly, it would be hard to answer the question, "which boards support MCUBoot: yes/no" or "which boards support MCUBoot w/ swap?"

As a reference, there's a similar problem for web/javascript developers and as a result https://caniuse.com/ & https://kangax.github.io/compat-table/es6/ were created. The former is manually curated while the latter is generated from standardized tests.

The main question is how can some or all the "does this do x" information be collected? Is there a template we need to ask each author to fill out and generate a report? Or does some or most of that information already exist and it's just a matter of creating scripts and .rst templates?

**Comment by mbolivar-nordic** [comment]
This is a longstanding problem. The plan of record is to draw a lot of the answers from devicetree. This will cover things like what peripherals are supported. Several attempts have been made and nothing has worked yet.

So yes, it is a matter of creating scripts and .rst templates, but it's a big problem and I wouldn't say it's "just a matter" except perhaps with a heavy wink.
